File: skypydb/database/mixins/vector/collections/count_items.py
"""Collection counting mixin for vector storage."""

from skypydb.security.validation import InputValidator


class CountItemsMixin:
    """Provide item count behavior per vector collection."""

    def count(self, collection_name: str) -> int:
        """Return the number of rows in a collection."""

        collection_name = InputValidator.validate_table_name(collection_name)
        if not self.collection_exists(collection_name):
            raise ValueError(f"Collection '{collection_name}' not found")

        cursor = self.conn.cursor()
        cursor.execute(f"SELECT COUNT(*) FROM [vec_{collection_name}]")
        return cursor.fetchone()[0]

File: skypydb-js/src/database/operations/vector/collections/count_items.ts
import { CollectionNotFoundError } from "../../../../errors";
import { InputValidator } from "../../../../security/validation";
import { GetCollectionMixin } from "./get_collection";

export abstract class CountItemsMixin extends GetCollectionMixin {
  count(collection_name: string): number {
    const validated = InputValidator.validate_table_name(collection_name);
    if (!this.collection_exists(validated)) {
      throw new CollectionNotFoundError(`Collection '${validated}' not found`);
    }

    const row = this.conn
      .prepare(`SELECT COUNT(*) AS count FROM [vec_${validated}]`)
      .get() as { count: number };
    return Number(row.count);
  }
}
